plugins/techdocs/src/reader/components/TechDocsReaderPage/TechDocsReaderPage.tsx

```diff
@@ -97,11 +97,9 @@ export const TechDocsReaderPage = (props: TechDocsReaderPageProps) => {
     });
 
     return (
-      (page as JSX.Element) || (
-        <TechDocsReaderPageProvider entityRef={entityRef}>
-          <TechDocsReaderLayout />
-        </TechDocsReaderPageProvider>
-      )
+      <TechDocsReaderPageProvider entityRef={entityRef}>
+        {(page as JSX.Element) || <TechDocsReaderLayout />}
```

Comment:
Maybe there will never be situation where you customize your reader page so much that you don't even want to use the reader page provider, but I guess thats why it was put outside of it before. Did that cause issues?

Reply:
**Warning**: My understanding of the problem is evolving/growing as I try and explain it.  Here's my best explanation right now!

The problem comes if your custom reader page uses any of the existing components (like `<TechDocsReaderPageContent />`), which all depend on the reader page provider.  Without it: docs won't load at all and it just waits infinitely (see screenshot in the PR description).

Previously, this problem was hidden because the components were indirectly being wrapped in a reader page provider: the custom app and documentation used `<TechDocsReader>` both as the route element _and_ the wrapper around the outlet.  So the first time through `<TechDocsReader />`, it had no children, but it had an outlet, so it rendered the outlet.  The outlet contained another `<TechDocsReader />`, but this time with children, so it rendered the children within the provider.

This construction isn't great, not only because it kind of hides the underlying dependency, but also because `<TechDocsReader />` is a routable extension, which aren't intended to be rendered outside of `<Route>` elements in the main app.  ...It happens to work in the example app, but it may not work in more complex setups (...Spotify's Backstage/TechDocs happens to fall into this category 😬 ).  You'll end up with an `Uncaught Error: Routable extension component was not discovered in the app element tree. Routable extension components may not be rendered by other components and must be directly available as an element within the App provider component.`

The alternative would be to keep this code how it was, but add documentation that, if you want to use the core reader page components, you have to wrap them in a `<TechDocsReaderPageProvider />`.  ...But that feels like unnecessary extra steps, at the moment, given how there aren't really any viable alternatives to `<TechDocsReaderPageContent />` yet.

Regarding what's actually there in the provider: it seems like the entity will always be relevant, and perhaps the TechDocs metadata would also always be relevant (in some abstract form).  ...I see where the shadowRoot + setter might be an unfortunate thing to always include.  But at least there's no performance overhead in including it.

Reply:
Yeah I think wrapping even a custom page in the `TechDocsReaderPageProvider ` is for now the best way forward. There will of course be situations where you can use the header and your custom content component for example, so it could be valuable to wrap each of the components that can be used to compose your reader page in the context it consumes. But we can re-evaluate this in the near future and focus on fixing the bugs for now!